Libs/vtkAddon/vtkPersonInformation.cxx

```diff
@@ -0,0 +1,175 @@
+{
+  if (!source)
+    {
+    return;
+    }
+  this->Data = source->Data;
+}
+
+//----------------------------------------------------------------------------
+void vtkPersonInformation::SetCustomString(const std::string& key, const std::string& value)
+{
+  std::string currentValue = this->Data[key];
+  if (value.compare(currentValue) == 0)
+    {
+    // no change
+    return;
+    }
+  this->Data[key] = value;
+  this->Modified();
+}
+
+//----------------------------------------------------------------------------
+std::string vtkPersonInformation::GetCustomString(const std::string& key)
+{
+  std::map<std::string, std::string>::iterator it = this->Data.find(key);
+  if (it == this->Data.end())
+    {
+    return "";
+    }
+  return it->second;
+}
+
+//----------------------------------------------------------------------------
+std::string vtkPersonInformation::WriteToString()
+{
+  std::string output;
+  for (std::map<std::string, std::string>::iterator it = this->Data.begin();
+    it != this->Data.end(); ++it)
+    {
+    if (!output.empty())
+      {
+      output += ";";
+      }
+    output += this->EncodeString(it->first) + ":" + this->EncodeString(it->second);
+    }
+  return output;
+}
+
+//----------------------------------------------------------------------------
+bool vtkPersonInformation::ReadFromString(const std::string& data)
+{
+  this->Data.clear();
+  std::stringstream keyValuePairs(data);
+  std::string keyValuePair;
+  while (std::getline(keyValuePairs, keyValuePair, ';'))
+    {
+    int colonIndex = keyValuePair.find(':');
+    std::string key = this->DecodeString(keyValuePair.substr(0, colonIndex));
+    std::string value = this->DecodeString(keyValuePair.substr(colonIndex + 1));
+    this->SetCustomString(key, value);
+    }
+  return true;
+}
+
+//----------------------------------------------------------------------------
+std::string vtkPersonInformation::EncodeString(const std::string& value)
+{
+  std::string output = value;
+  // encode percent sign and semicolon (semicolon is a special character because it separates attributes)
+  vtksys::SystemTools::ReplaceString(output, "%", "%25");
+  vtksys::SystemTools::ReplaceString(output, ":", "%3A");
+  vtksys::SystemTools::ReplaceString(output, ";", "%3B");
+  return output;
+}
+
+//----------------------------------------------------------------------------
+std::string vtkPersonInformation::DecodeString(const std::string& value)
+{
+  std::string output = value;
+  // decode percent sign and semicolon (semicolon is a special character because it separates attributes)
+  vtksys::SystemTools::ReplaceString(output, "%3B", ";");
+  vtksys::SystemTools::ReplaceString(output, "%3A", ":");
+  vtksys::SystemTools::ReplaceString(output, "%25", "%");
+  return output;
+}
+
+//----------------------------------------------------------------------------
+bool vtkPersonInformation::IsEmailValid(const std::string& value)
```

Comment:
And since `kwsys` provide a [RegularExpression](https://gitlab.kitware.com/utils/kwsys/blob/master/RegularExpression.hxx.in) class, if needed, this type of check could be more sophisticated ... 

and by optionally associating regex if each field, UI could leverage them ...